src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { css } from 'styled-components';
+
+import { Box, Icon, Text } from '@/components';
+import { HeadingBlock, useEditorStore } from '@/features/docs/doc-editor';
+import { MAIN_LAYOUT_ID } from '@/layouts/conf';
+
+import { Heading } from './Heading';
+
+type DocSummaryProps = {
+  headings: HeadingBlock[];
+};
+
+export const DocSummary = ({ headings }: DocSummaryProps) => {
+  const { editor } = useEditorStore();
+
+  const [headingIdHighlight, setHeadingIdHighlight] = useState<string>();
+
+  const { t } = useTranslation();
+  const [isHover, setIsHover] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (!headings) {
+        return;
+      }
+
+      for (const heading of headings) {
+        const elHeading = document.body.querySelector(
+          `.bn-block-outer[data-id="${heading.id}"] [data-content-type="heading"]:first-child`,
+        );
+
+        if (!elHeading) {
+          return;
+        }
+
+        const rect = elHeading.getBoundingClientRect();
+        const isVisible =
+          rect.top + rect.height >= 1 &&
+          rect.bottom <=
+            (window.innerHeight || document.documentElement.clientHeight);
+
+        if (isVisible) {
+          setHeadingIdHighlight(heading.id);
+          break;
+        }
+      }
+    };
+
+    document.getElementById(MAIN_LAYOUT_ID)?.addEventListener('scroll', () => {
+      setTimeout(() => {
+        handleScroll();
+      }, 300);
+    });
+
+    handleScroll();
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, [headings, setHeadingIdHighlight]);
+
+  if (!editor) {
+    return null;
+  }
+
+  return (
+    <Box
+      onMouseEnter={() => {
+        setIsHover(true);
+        setTimeout(() => {
+          const element = document.getElementById(
+            `heading-${headingIdHighlight}`,
+          );
+
+          element?.scrollIntoView({
+            behavior: 'smooth',
+            inline: 'center',
+            block: 'center',
+          });
```

Comment:
The hover is not on the element where you clicked:

[scrnli_5OjEqYWaR2Im4U.webm](https://github.com/user-attachments/assets/94b6e87d-049e-42d1-a766-13b21ef6745e)